lib/ProductOpener/Images.pm

```diff
@@ -1349,4 +1349,132 @@ HTML
 	return $html;
 }
 
+
+sub extract_text_from_image($$$$) {
+
+	my $product_ref = shift;
+	my $id = shift;
+	my $field = shift;
+	my $ocr_engine = shift;
+
+	delete $product_ref->{$field};
+
+	my $path = product_path($product_ref->{code});
+	my $status = 1;
+
+	my $filename = '';
+
+	my $lc = $product_ref->{lc};
+
+	if ($id =~ /_(\w\w)$/) {
+		$lc = $1;
+	}
+
+	my $size = 'full';
+	if ((defined $product_ref->{images}) and (defined $product_ref->{images}{$id})
+		and (defined $product_ref->{images}{$id}{sizes}) and (defined $product_ref->{images}{$id}{sizes}{$size})) {
+		$filename = $id . '.' . $product_ref->{images}{$id}{rev} ;
+	}
```

Comment:
Shouldn't we have an `else` clause for the case that `$filename` cannot be set? The following code that uses `$image` or `$image_url` would be useless in that case, because the URLs would only point to non-existing images.

Reply:
Agreed, we should bail out.

Reply:
Added else clause.